pysteps/blending/steps.py

```diff
@@ -0,0 +1,1746 @@
+    )
+
+    # determine the precipitation threshold mask
+    if conditional:
+        MASK_thr = np.logical_and.reduce(
+            [R[i, :, :] >= R_thr for i in range(R.shape[0])]
+        )
+    else:
+        MASK_thr = None
+
+    ###
+    # 1. Start with the radar rainfall fields. We want the fields in a
+    # Lagrangian space
+    ###
+    # advect the previous precipitation fields to the same position with the
+    # most recent one (i.e. transform them into the Lagrangian coordinates)
+    extrap_kwargs = extrap_kwargs.copy()
+    extrap_kwargs["xy_coords"] = xy_coords
+    res = list()
+
+    def f(R, i):
+        return extrapolator_method(
+            R[i, :, :],
+            V,
+            ar_order - i,
+            "min",
+            allow_nonfinite_values=True,
+            **extrap_kwargs,
+        )[-1]
+
+    for i in range(ar_order):
+        if not DASK_IMPORTED:
+            R[i, :, :] = f(R, i)
+        else:
+            res.append(dask.delayed(f)(R, i))
+
+    if DASK_IMPORTED:
+        num_workers_ = len(res) if num_workers > len(res) else num_workers
+        R = np.stack(list(dask.compute(*res, num_workers=num_workers_)) + [R[-1, :, :]])
+
+    # replace non-finite values with the minimum value
+    R = R.copy()
+    for i in range(R.shape[0]):
+        R[i, ~np.isfinite(R[i, :])] = np.nanmin(R[i, :])
+
+    ###
+    # 2. Initialize the noise method
+    ###
+    if noise_method is not None:
+        # get methods for perturbations
+        init_noise, generate_noise = noise.get_method(noise_method)
+
+        # initialize the perturbation generator for the precipitation field
+        pp = init_noise(R, fft_method=fft, **noise_kwargs)
+
+        if noise_stddev_adj == "auto":
+            print("Computing noise adjustment coefficients... ", end="", flush=True)
+            if measure_time:
+                starttime = time.time()
+
+            R_min = np.min(R)
+            noise_std_coeffs = noise.utils.compute_noise_stddev_adjs(
+                R[-1, :, :],
+                R_thr,
+                R_min,
+                filter,
+                decomp_method,
+                pp,
+                generate_noise,
+                20,
+                conditional=True,
+                num_workers=num_workers,
+            )
+
+            if measure_time:
+                print("%.2f seconds." % (time.time() - starttime))
+            else:
+                print("done.")
+        elif noise_stddev_adj == "fixed":
+            f = lambda k: 1.0 / (0.75 + 0.09 * k)
+            noise_std_coeffs = [f(k) for k in range(1, n_cascade_levels + 1)]
+        else:
+            noise_std_coeffs = np.ones(n_cascade_levels)
+
+        if noise_stddev_adj is not None:
+            print("noise std. dev. coeffs:   %s" % str(noise_std_coeffs))
+
+    ###
+    # 3. Perform the cascade decomposition for the input precip fields and
+    # The decomposition for the (NWP) model fields is already present
```

Comment:
The reason for that is that the decomposition of the NWP forecast takes considerably more time than a couple of radar images, especially when the NWP forecast has a fine temporal resolution (for instance 5 minutes for the RMI NWP forecasts) and a forecast horizon of 1 or 2 days.
As a new NWP forecast is not issued every 5 minutes, we thought it would be an unnecessary computational job to recalculate the NWP velocity fields and decompose it for every new nowcasts. Instead, we decided to reproject the NWP and calculate it beforehand, with the option to store the information and continue with the blended forecast.

Reply:
Oh yes now I remember!

OK I propose a compromise then: offer the option to provide the preprocessed NWP forecast, otherwise these are compute inside the method, what do you think?

Reply:
That sounds like a good option! I'll try to implement it.